src/core/processing/qgsprocessingalgorithm.h

```diff
@@ -1022,6 +1022,17 @@ class CORE_EXPORT QgsProcessingAlgorithm
      */
     static QString invalidSinkError( const QVariantMap &parameters, const QString &name );
 
+    /**
+     * Returns a user-friendly string to use as an error when a feature cannot be
+     * written into a sink.
+     *
+     * The \a parameters argument should give the algorithms parameter map, and the \a name
+     * should correspond to the sink parameter name.
+     *
+     * \since QGIS 3.22
+     */
+    static QString writeFeatureError( const QVariantMap &parameters, const QString &name );
```

Comment:
Is it too much to ask for a QgsFeatureSink pointer here as well? We could potentially add a "lastError" to QgsFeatureSink later and get a descriptive reason why the add failed...

Reply:
done